### PR TITLE
rename ocaml interface filetype

### PR DESCRIPTION
**Pull request by aspeddro**
Syncs with the filetypes detected by LSP.

https://github.com/neovim/nvim-lspconfig/blob/da7461b596d70fa47b50bf3a7acfaef94c47727d/lua/lspconfig/server_configurations/ocamllsp.lua#L19

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -429,7 +429,7 @@ list.ocaml_interface = {
     location = "interface",
   },
   maintainers = { "@undu" },
-  filetype = "ocamlinterface",
+  filetype = "ocaml.interface",
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:432]: Can you add the old name to `filetype_to_parsername` table?

File: lua/nvim-treesitter/parsers.lua [hunk 2]
```diff
@@ -18,6 +18,7 @@ local filetype_to_parsername = {
   pandoc = "markdown",
   rmd = "markdown",
   cs = "c_sharp",
+  ["ocaml.interface"] = "ocamlinterface",
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:21]: but the filetype is `ocamlinterface` and parser name is `ocaml_interface`

File: lua/nvim-treesitter/parsers.lua [hunk 3]
```diff
@@ -18,6 +18,7 @@ local filetype_to_parsername = {
   pandoc = "markdown",
   rmd = "markdown",
   cs = "c_sharp",
+  ["ocaml.interface"] = "ocaml_interface",
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:21]: you mean `["ocamlinterface"] = "ocaml_interface"`

Comment (theHamsta) [reply lua/nvim-treesitter/parsers.lua:21]: `["ocam.linterface"] = "ocaml_interface" was already mentioned below. `ocaml_interface` is the name in the parser C binary

Comment (aspeddro) [reply lua/nvim-treesitter/parsers.lua:21]: The filetype detected by LSP is `ocaml.interface`

```lua
filetypes = { 'ocaml', 'ocaml.menhir', 'ocaml.interface', 'ocaml.ocamllex', 'reason', 'dune' }
```

Comment (theHamsta) [reply lua/nvim-treesitter/parsers.lua:21]: but the filetype that was here before was `ocamlinterface`

Comment (undu) [reply lua/nvim-treesitter/parsers.lua:21]: When I made the chanes I checked that Vim didn't allow for filetypes to have that dot, hence the type in LSP is different from the vim filetype. I'm not sure what's the current situation.

Comment (theHamsta) [reply lua/nvim-treesitter/parsers.lua:21]: Vim will fallback to `ocaml` with `ocaml.interface` has not traditional highlighting. I think this behavior was also implemented for nvim-treesitter though we typically recognize the string as a whole for most parsers.

Should we only keep the new filetype? Or will there be users with `ocamlinterface` configured?

Comment (undu) [reply lua/nvim-treesitter/parsers.lua:21]: Ale is configured to use ocamlinterface, and do the conversion: https://github.com/dense-analysis/ale/blob/a02a4f2811f810877f3c3859cca963f9578ff94a/autoload/ale/handlers/ocamllsp.vim#L7

I opened several PRs to follow that convention, detailed here: https://github.com/ocaml/vim-ocaml/pull/61

IMO changing lspconfig to follow that convention is easier than doing the reverse

Comment (aspeddro) [reply lua/nvim-treesitter/parsers.lua:21]: Closing because the LSP setup does the conversion ocamlinterface => 'ocaml.interface'

**Pull request closed without merging**